receiver/elasticapmreceiver/receiver.go

```diff
@@ -236,18 +246,59 @@ func (r *elasticAPMReceiver) processBatch(ctx context.Context, batch *modelpb.Ba
 			ss := rs.ScopeSpans().AppendEmpty()
 			s := ss.Spans().AppendEmpty()
 
+			traceId, err := traceIDFromHex(event.Trace.Id)
+			if err == nil {
```

Comment:
I would keep the receiver code very small, and extract all the logic into internal methods, so they can be more easily tested. But maybe that's my preference.

Reply:
Yeah, I agree. I moved out all these lines into separate functions. One nit here: https://github.com/elastic/opentelemetry-collector-components/pull/294/files#r1963186038 - but I feel it's ok.